test.js

```diff
@@ -12,3 +12,144 @@ describe('countToNum', () => {
+    })
+})
+
+//numberOfVowels
+describe('numberOfVowels', () => {
+    it('Returns the number 2 for the string "apple"', () => {
+        expect(numberOfVowels('apple')).to.eq(2)
+    })
+    it('Returns the number 1 for the string "dog"', () => {
+        expect(numberOfVowels('dog')).to.eq(1)
+    })
+})
+
+//separateNames
+describe('separateNames', () => {
+    it('Returns ["Ben", "Bradfield"] from the input of "ben bradfield"', () => {
+        expect(separateNames('ben bradfield')).to.deep.eq(['Ben', 'Bradfield'])
+    })
+    it('Returns ["Rob", "Swanson"] from the input of "rob swanson"', () => {
+        expect(separateNames('ben bradfield')).to.deep.eq(['Ben', 'Bradfield'])
+    })
+})
+
+//mostFrequentLetter
+describe('mostFrequentLetter', () => {
+    it('Returns the most frequent letter and the count. Ex. "a, 6" from "aaaaaa"', () => {
+        expect(mostFrequentLetter('aaaaaa')).to.eq('a, 6')
+    })
+    it('Returns the most frequent letter and the count. Ex. "n, 3" from "cinnimon"', () => {
+        expect(mostFrequentLetter('cinnimon')).to.eq('n, 3')
+    })
+})
+
+//palindrome
+describe('palindrome', () => {
+    it('Returns true if the input string is a palindrome', () => {
+        expect(palindrome('hannah')).to.equal(true)
+    })
+    it('Returns false if the input string is not a palindrome', () => {
+        expect(palindrome('ben')).to.equal(false)
+    })
+})
+
+//calculator
+describe('calculator', () => {
+    it('Returns the number 5', () => {
+        expect(calculator('add',2,3)).to.eq(5)
+    })
+    it('Returns the number 10', () => {
+        expect(calculator('multiply',5,2)).to.eq(10)
+    })
+})
+
+//birthday
+describe('birthday', () => {
+    it('Returns the string "hi my name is [name] and my birthday is [birthday]" if the person exists in the object', () => {
+        expect(birthday('bob')).to.eq('hi my name is bob and my birthday is June 8')
+    })
+    it('Returns the string "this person does not exist" if the person does not exist in the object', () => {
+        expect(birthday('ryan')).to.eq('this person does not exist')
+    })
+})
+
+//iterateBirthday
+// describe('iterateBirthday', () => {
+//     it('Returns the string "hi my name is [name] and my birthday is [birthday]" for each person in the object', () => {
+//         expect(iterateBirthday(myFriends)).to.eq(//console.logs?)
+//     })
+// })
+
+//highestAverage
+describe('highestAverage', () => {
+    it('Returns the string "[name] has the highest average with an average of [averageTestScore]"', () => {
+        expect(highestAverage(myFriendsScores)).to.eq("joel has the highest average with an average of 89.25")
+    })
+})
+
+//addAverageTestScore
+describe('addAverageTestScore', () => {
+    it('Returns the same object but adds the key "average" with its value of the persons average test score', () => {
+        expect(addAverageTestScore(myFriendsScores)).to.deep.eq({"ryan": {
```

Comment:
maybe makes more sense to just test the average components of this object, not the whole object